docs/COREDUMP.md

```diff
@@ -0,0 +1,148 @@
+---
+title: systemd Coredump Handling
+category: Concepts
+layout: default
+SPDX-License-Identifier: LGPL-2.1-or-later
+---
+
+# systemd Coredump Handling
+
+## Support in the Service Manager (PID 1)
+
+The systemd service manager natively provides coredump handling functionality,
+as implemented by the Linux kernel. Specifically, PID 1 provides the following
+functionality:
+
+1. During very early boot it will raise the
+   [`LIMIT_CORE`](https://man7.org/linux/man-pages/man2/getrlimit.2.html)
+   resource limit for itself to infinity (and thus implicitly also all its
+   children). This removes any limits on the size of generated coredumps, for
+   all invoked processes, from earliest boot on. (The Linux kernel sets the
+   limit to 0 by default.)
+
+2. At the same time it will turn off coredump handling in the kernel by writing
+   `|/bin/false` into `/proc/sys/kernel/core_pattern` (see
+   [core(5)](https://man7.org/linux/man-pages/man5/core.5.html) for
+   details). This means that coredumps are not actually accepted by
+   userspace. (The linux kernel sets the pattern to `core` by default, so that
+   coredumps are written to the current working directory of the crashing
+   process.)
+
+Net effect: by default coredumps are disabled, but by means of the
+`core_pattern` rather than by size limit. This is generally preferable, since
+the pattern can be updated trivially at the right time to enable coredumping
+once the system is ready, taking comprehensive effect on all userspace. (Or to
+say this differently: disabling coredumps via the size limit is problematic,
+since it cannot easily be undone without iterating through all already running
+processes once the system is ready for coredump handling.)
+
+By updating the `core_pattern` at an appropriate point during the boot,
+coredump processing may be enabled. This moment can precisely be scheduled: any
+coredumps happening before that point are suppressed, coredumps happening after
+are processed.
```

Comment:
"Processing of core dumps may be enabled at the appropriate time by updating the `kernel.core_pattern` sysctl. Only coredumps that happen later will be processed."